### ISSUE TITLE
Number of steps does not match batch size

### ISSUE THREAD

**Issue by Pashahlis** [kohya-ss/musubi-tuner#446]
I am trying out a higher batch size than 1 for the first time in forever, and set it to 18 aka my amount of training images.

Before that, I always ran 1 batch size and 1800 steps. So I expected the steps to be 100 this time.

But instead I am getting 700 steps or 7 steps per epoch, instead of the expected 100 and 1 respectively, which does not seem to make any sense to me?

`INFO:musubi_tuner.dataset.image_video_dataset:bucket: (640, 1440), count: 1
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (720, 1280), count: 1
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (832, 1104), count: 9
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (848, 1072), count: 1
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (960, 960), count: 2
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (1104, 832), count: 2
INFO:musubi_tuner.dataset.image_video_dataset:bucket: (1280, 720), count: 2
INFO:musubi_tuner.dataset.image_video_dataset:total batches: 7
INFO:musubi_tuner.hv_train_network:preparing accelerator
accelerator device: cuda
INFO:musubi_tuner.hv_train_network:DiT precision: torch.bfloat16, weight precision: torch.bfloat16
INFO:musubi_tuner.hv_train_network:Loading DiT model from /workspace/musubi-tuner/models/diffusion_models/split_files/diffusion_models/qwen_image_bf16.safetensors
INFO:musubi_tuner.qwen_image.qwen_image_model:Creating QwenImageTransformer2DModel
INFO:musubi_tuner.qwen_image.qwen_image_model:Loading DiT model from /workspace/musubi-tuner/models/diffusion_models/split_files/diffusion_models/qwen_image_bf16.safetensors, device=cuda
INFO:musubi_tuner.utils.lora_utils:Loading model files: ['/workspace/musubi-tuner/models/diffusion_models/split_files/diffusion_models/qwen_image_bf16.safetensors']
INFO:musubi_tuner.utils.lora_utils:Loading state dict without FP8 optimization. Dtype of weight: torch.bfloat16, hook enabled: False
INFO:musubi_tuner.qwen_image.qwen_image_model:Loaded DiT model from /workspace/musubi-tuner/models/diffusion_models/split_files/diffusion_models/qwen_image_bf16.safetensors, info=<All keys matched successfully>
import network module: networks.lora_qwen_image
INFO:musubi_tuner.networks.lora:create LoRA network. base dim (rank): 32, alpha: 32.0
INFO:musubi_tuner.networks.lora:neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
INFO:musubi_tuner.networks.lora:create LoRA for U-Net/DiT: 840 modules.
INFO:musubi_tuner.networks.lora:enable LoRA for U-Net: 840 modules
QwenModel: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:musubi_tuner.hv_train_network:use AdamW optimizer | {'weight_decay': 0.1}
override steps. steps for 100 epochs is / 指定エポックまでのステップ数: 700
running training / 学習開始
  num train items / 学習画像、動画数: 18
  num batches per epoch / 1epochのバッチ数: 7
  num epochs / epoch数: 100
  batch size per device / バッチサイズ: 18
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 700
INFO:musubi_tuner.hv_train_network:set DiT model name for metadata: /workspace/musubi-tuner/models/diffusion_models/split_files/diffusion_models/qwen_image_bf16.safetensors
INFO:musubi_tuner.hv_train_network:set VAE model name for metadata: /workspace/musubi-tuner/models/vae/vae/diffusion_pytorch_model.safetensors
steps:   0%|                                                                                                                                   | 0/700 [00:00<?, ?it/s]INFO:musubi_tuner.hv_train_network:DiT dtype: torch.bfloat16, device: cuda:0

epoch 1/100
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   1%|█                                                                                                         | 7/700 [00:17<28:16,  2.45s/it, avr_loss=0.0792]
epoch 2/100`

**Comment by seruva19** [comment]
Your images have different sizes, so they are split into 7 buckets. The largest bucket only has 9 images, which is still fewer than your batch size of 18. Since batches can't mix across buckets, the effective number of batches per epoch is 7. Therefore, the total number of steps is num_batches_per_epoch * num_epochs = 7 * 100 = 700.
